geode-core/src/main/java/org/apache/geode/distributed/internal/InternalLocator.java

```diff
@@ -773,15 +773,19 @@ private void startCache(DistributedSystem system) throws IOException {
     startClusterManagementService();
   }
 
-  private void startClusterManagementService() throws IOException {
+  @VisibleForTesting
+  void startClusterManagementService() throws IOException {
     startConfigurationPersistenceService();
 
-    if (internalCache == null) {
+    InternalCache myCache = this.internalCache;
+    InternalLocator currentLocator = InternalLocator.locator;
+
+    if (myCache == null || currentLocator == null) {
       return;
     }
 
-    clusterManagementService = new LocatorClusterManagementService(locator.internalCache,
-        locator.configurationPersistenceService);
+    clusterManagementService = new LocatorClusterManagementService(currentLocator.internalCache,
```

Comment:
Since this is an instance method on InternalLocator, why don't we just use this.internalCache and this.configurationPersistenceService instead of trying to get the locator from a static. This method should be initializing itself.

Reply:
yes, I agree.  I'm going to make a pass through InternalLocator and remove the use of the "locator" static wherever possible.  These uses all seem to come from cluster-configuration work.